### PR TITLE
Support durations in aggregation-functions

### PR DESCRIPTION
**Pull request by FlorentinD**
Cypher supports the aggregation of `Durations`, however Spark does not support the aggregation of `CalendarIntervals`.

This will allow the aggregation of `CalendarIntervals` via user-defined aggregation functions (_udafs_).

### CODE REVIEW

File: morpheus-spark-cypher/src/main/scala/org/opencypher/morpheus/impl/temporal/TemporalUdafs.scala [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.expressions.{MutableAggregationBuffer, UserDefinedAggregateFunction}
+import org.apache.spark.sql.types.{CalendarIntervalType, DataType, LongType, StructField, StructType}
+import org.apache.spark.unsafe.types.CalendarInterval
+import org.opencypher.okapi.impl.temporal.TemporalConstants
+import org.opencypher.morpheus.impl.temporal.TemporalConversions._
+
+object TemporalUdafs extends Logging{
+
+  abstract class SimpleDurationAggregation(aggrName : String) extends UserDefinedAggregateFunction {
```
Comment (DarthMax) [inline morpheus-spark-cypher/src/main/scala/org/opencypher/morpheus/impl/temporal/TemporalUdafs.scala]:
```suggestion
  abstract class SimpleDurationAggregation(aggrName: String) extends UserDefinedAggregateFunction {
```

File: morpheus-spark-cypher/src/main/scala/org/opencypher/morpheus/impl/temporal/TemporalUdfs.scala [hunk 2]
```diff
@@ -30,8 +30,10 @@ import java.sql.{Date, Timestamp}
 import java.time.temporal.{ChronoField, IsoFields, TemporalField}
 
 import org.apache.logging.log4j.scala.Logging
-import org.apache.spark.sql.expressions.UserDefinedFunction
+import org.apache.spark.sql.Row
```
Comment (DarthMax) [inline morpheus-spark-cypher/src/main/scala/org/opencypher/morpheus/impl/temporal/TemporalUdfs.scala]: I think these imports can be removed again

File: morpheus-spark-cypher/src/main/scala/org/opencypher/morpheus/impl/temporal/TemporalUdafs.scala (hunk 1 above)
Comment (FlorentinD) [reply morpheus-spark-cypher/src/main/scala/org/opencypher/morpheus/impl/temporal/TemporalUdafs.scala]: Solved via auto-format

File: morpheus-spark-cypher/src/main/scala/org/opencypher/morpheus/impl/temporal/TemporalUdfs.scala (hunk 2 above)
Comment (FlorentinD) [reply morpheus-spark-cypher/src/main/scala/org/opencypher/morpheus/impl/temporal/TemporalUdfs.scala]: Right, I forgot

**Pull request merged**